src/backend/access/aocs/aocsam_handler.c

```diff
@@ -1174,7 +1174,14 @@ aoco_tuple_lock(Relation relation, ItemPointer tid, Snapshot snapshot,
 static void
 aoco_finish_bulk_insert(Relation relation, int options)
 {
-	aoco_dml_finish(relation, CMD_INSERT);
+	/*
+	 * https://github.com/cloudberrydb/cloudberrydb/issues/547
+	 * Do not clean up context or resource here(ex: aoco_dml_finish),
+	 * finish_bulk_insert routine will be called more than once during
+	 * COPY FROM if the partition buffer is flushed but COPY is not finished.
+	 *
+	 * FIXME: Do we need to keep this am routine? Seems no used anymore.
```

Comment:
https://github.com/cloudberrydb/cloudberrydb/blob/main/src/backend/commands/copyfrom.c#L2534
Remove the call `table_finish_bulk_insert`?

Reply:
+1, that call is useless as we have already call related functions like aoco_dml_finish() 
https://github.com/cloudberrydb/cloudberrydb/blob/main/src/backend/commands/copyfrom.c#L2433
I think we are ok to  remove it.

Reply:
Done.